### ISSUE TITLE
Allow us to set Display names in server context.

### ISSUE THREAD

**Issue by andrewkm** [lucko/LuckPerms#963]
Previously, I was using custom assignments to do the following:
```
  rule-defaults:
  if:
    has-true: <group.default>
  give:
    - skyblock/group.skyblock
    - lobby/group.lobby
    - vmr/group.vmr
    - kitpvp/group.kitpvp
```
This would allow me to weight those groups higher than default, and add a display name to each group. **However** this takes away from storage optimizations as it saves **every single default user**.

I've since moved to removing the custom assignment and using the following default group instead:
```
/lp group default parent add vmr server=vmr
/lp group default parent add skyblock server=skyblock
/lp group default parent add main server=main
/lp group default parent add lobby server=lobby
/lp group default parent add kitpvp server=kitpvp
```
This allows default users who come and go to not take a toll on our DB size, however we have no way to assign display name per server for these defaults.

Say I want defaults on Skyblock to be called Pilots
And I want defaults on Lobby to be called Travelers.

This is not possible as there is no way to:
/lp group skyblock setdisplayname Pilot server=skyblock

I can only:
/lp group skyblock setdisplayname Pilot

However with the above, users on the Lobby are called Pilots, when I would prefer them to be called Travelers.

Basically, we would like to know if there is a way, or if you can please allow a way to set displayname with a server-context.

Our goal is for this to work:
```
/lp group default parent add vmr server=vmr
/lp group default parent add skyblock server=skyblock
/lp group default parent add main server=main
/lp group default parent add lobby server=lobby
/lp group default parent add kitpvp server=kitpvp
/lp group default permission set displayname.Runner true server=vmr
/lp group default permission set displayname.Pilot true server=skyblock
/lp group default permission set displayname.Builder true server=main
/lp group default permission set displayname.Traveler true server=lobby
/lp group default permission set displayname.Fighter true server=kitpvp
```

Any help would be really appreciated.

**Comment by andrewkm** [comment]
Thanks so much @lucko tested and works perfectly! <3
